tests/test_engine.py

```diff
@@ -103,3 +103,37 @@ async def test_engine_execution_works(engine: Engine) -> None:
         }
     )
     assert_frame_equal(res, expected)
+
+
+async def test_sut_utilize_factor_arguments() -> None:
```

Comment:
```suggestion
async def test_sut_utilizes_factor_arguments() -> None:
```

sut 보다 위에서 쓴 engine이라는 말을 써주는게 좀 더 어떤걸 테스트하는지 알기 쉽지 않을까 싶네요.